HousingRepairsOnlineApi/UseCases/SendAppointmentConfirmationSmsEmailUseCase.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Ardalis.GuardClauses;
+using HousingRepairsOnlineApi.Domain;
+using HousingRepairsOnlineApi.Gateways;
+
+namespace HousingRepairsOnlineApi.UseCases
+{
+    public class SendAppointmentConfirmationEmailUseCase : ISendAppointmentConfirmationEmailUseCase
+    {
+        private readonly INotifyGateway notifyGateway;
+        private readonly string templateId;
+
+        public SendAppointmentConfirmationEmailUseCase(INotifyGateway notifyGateway, string templateId)
+        {
+            this.notifyGateway = notifyGateway;
+            this.templateId = templateId;
+        }
+        public async Task<SendEmailConfirmationResponse> Execute(string email, string bookingRef, string appointmentTime)
+        {
+            Guard.Against.NullOrWhiteSpace(email, nameof(email), "The email provided is invalid");
+            Guard.Against.NullOrWhiteSpace(bookingRef, nameof(bookingRef), "The booking reference provided is invalid");
+            Guard.Against.NullOrWhiteSpace(appointmentTime, nameof(appointmentTime), "The appointment time provided is invalid");
+
```

Comment:
Yeah this is something i thought about. The reason i dont have it is because I dont _think_ the regex is as clear for emails. So if someone had a funky email it may fail the validation, I didnt want the application to act as a gatekeeper. I thought it would have been better if it went though to notify and it failed there. The same may apply to phone numbers but i think the frontend validation prevents invalid phone numbers and Im not sure how prescriptive the frontend validation is for emails- it may just look for an @.

Reply:
Without handling the potential Notify library exception, our code should validate the email address.

The frontend regex is [here](https://github.com/City-of-Lincoln-Council/housing-repairs-online-frontend/blob/main/helpers/validators.js#L29).